### DISCUSSION TITLE
GLSL Roadmap

### DISCUSSION THREAD

**Discussion by ferrellsl** [Joshua-Riek/ubuntu-rockchip#71] [Q&A]
Has anyone seen a roadmap about future GLSL updates/support for the Pi 5?  I'm porting various games to the Pi 5 but the current version of GLSL on the Pi 5 is somewhat behind what is needed to port some of theses games such as Quake2XP.  Quake2XP requires GLSL v 4.30-4.50.  My Pi 5 reports that it supports up to GLSL v 3.10.  I get a working binary when compiling Quake2XP but the graphics are broken due to outdated GLSL.

**Comment by Joshua-Riek** [comment]
Sometimes setting the GL version with an environment variable can help fix a few games `PAN_MESA_DEBUG=gofaster,gl3`.

But as for the GLSL roadmap, I have not looked into it much yet, but there is progress from mesa on the GPU drivers. But I think it's targeted for the 6x Linux kernel. I know the Rock Pi 5 community has been playing with 6.2 and these new GPU drivers, and it's still highly a work in progress. This is because recently the 6.2 Linux kernel was released with rk3588 support. But we will need to wait as 6.3 will include support for the rk3588s that the Orange Pi 5 uses.

**Comment by Fredobedo** [comment]
Dear @ferrellsl , 
it already looks very interesting all the games you ported.

Would it be possible to share something with us ? (i.e. Quake2 install for dummies :) )

Fred

**Reply by ferrellsl** [reply]
Install the required dependencies:
sudo apt install build-essential libopenal-dev libcurl-dev libsdl2-dev libvulkan-dev libglu1-mesa-dev freeglut3-dev mesa-common-dev libogg-dev


1. Clone the Yamagi quake2 repo:  git clone https://github.com/yquake2/yquake2
2. Move to the yquake2 folder:  cd yquake2
3. Run the make command:  make
4. If successful, there will now be a release folder in your yquake2 folder with all the necessary binaries.
5. Copy the contents of your quake2 baseq2 folder from your PC to the baseq2 folder in the release directory.
6. Run quake2 from the release folder by typing ./quake2


That's it.

**Comment by Joshua-Riek** [comment]
Someone created an extensive guide on how to setup box86, box64, and wine with mesa GPU driver for the rock pi5, I will check this out over the weekend and see what changes need to be made for the orange pi5.
https://github.com/neofeo/BOX86-BOX64-WINEx86-TUTORIAL
